stardao-mongodb/src/test/java/io/stardog/stardao/mongodb/AbstractMongoDaoTest.java

```diff
@@ -104,6 +119,19 @@ public void testFindByQuery() throws Exception {
     }
 
     @Test
+    public void testFindByQueryPartial() throws Exception {
+        TestUser created1 = dao.create(TestUser.builder().name("Ian").email("ian1@example.com").build());
+        TestUser created2 = dao.create(TestUser.builder().name("Ian").email("ian2@example.com").build());
+
+        Results<TestUser,ObjectId> results = dao.findByQuery(new Document("name", "Ian"), new Document("email", -1), new Document("name", 1));
```

Comment:
Here, `Results.of()` is copying existing `Iterable` object to immutable list using `copyOf()` which does it by iterating over all elements linearly. This means, results are not lazily loaded anymore. It's not going to be problematic for queries returning smaller result set but for queries returning thousands of documents, it will create memory pressure.

I am assuming that, `findByQuery()` is kind of intended for queries returning smaller result set ideally with some kind of pagination, right? In that case, this should be fine.